### PR TITLE
Deprecate strict mode

### PR DESCRIPTION
**Pull request by DRMacIver**
Hypothesis's strict mode is basically `warnings.simplefilter("error", HypothesisDeprecationWarning)` but with more confusing semantics.

This deprecates it. It is built on the work in #580 so will become simpler to review once that is merged. You can either ignore it until then or just look at the last few commits.

### CODE REVIEW

**Comment by DRMacIver** [comment]
the build on this is currently failing in ways I can't reproduce locally, which is annoying. :-(

**Comment by DRMacIver** [comment]
@HypothesisWorks/hypothesis-python-contributors after a long delay this is actually passing all its tests and ready for review.

File: tests/cover/test_database_usage.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def stuff():
                 st.binary(min_size=100),
                 lambda x: assume(not finicky) and has_a_non_zero_byte(x),
                 settings=settings(
```
Comment (Zac-HD) [inline tests/cover/test_database_usage.py]: No need to break over multiple lines now

File: RELEASE.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+RELEASE_TYPE: minor
+
+This release deprecates Hypothesis's strict mode, which turned Hypothesis's
+deprecation warnings into errors. Similar functionality can be achieved by
+using the standard library's warnings.simplefilter directly.
```
Comment (Zac-HD) [inline RELEASE.rst]: by using :func:`simplefilter('error', HypothesisDeprecationWarning) <python:warnings.simplefilter>` directly.

(I like explicit advice on how to update, and as many cross-references as possible)

**Pull request merged**